.github/workflows/unit-tests.yml

```diff
@@ -29,3 +29,7 @@ jobs:
           coverage run -m --source=src pytest -v tests/unit_test.py
           coverage=$(coverage report -m | tail -1 | tail -c 4 | head -c 2)
           if (( $coverage < 90 )); then echo "Coverage failed at ${coverage}%"; exit 1; else echo "Coverage passed, ${coverage}%"; fi
```

Comment:
On second thought, I think the code coverage report won't be uploaded if the previous step results in less than 90% coverage, exiting the workflow.

Reply:
That's a good catch! It should work now.